map/traffic_manager.cpp

```diff
@@ -93,3 +113,59 @@ void TrafficManager::CalculateSegmentsColoring(traffic::TrafficInfo const & traf
     output.emplace_back(sid, c.second);
   }
 }
+
+// static
+void TrafficManager::ThreadRoutine(TrafficManager * manager)
+{
+  CHECK(manager != nullptr, ());
+  while (manager->m_isRunning)
```

Comment:
m_isRunning is modified in destructor only, and is read in your auxiliary thread only. So, just protect it with your lock, and everything will be fine.